merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -234,6 +233,7 @@ private <Return> void stepEffectModel(
       this.tasks.put(task, progress.completedAt(currentTime, children));
       this.scheduledJobs.schedule(JobId.forTask(task), SubInstant.Tasks.at(currentTime));
     } else if (status instanceof TaskStatus.Delayed<Return> s) {
+      if (s.delay().isNegative()) throw new IllegalArgumentException("Cannot schedule a task in the past");
```

Comment:
Can we add this check to `SimulationEngine.scheduleTask` as well?

Reply:
I added an analogous check to `scheduleTask` - it's a mildly different situation, since the Simulation Engine doesn't know the current time (side note - should it?), so instead of saying "the past", I tried to give it a message that conveyed exactly what it was checking (and had different text from the other message in case we need to debug-by-grep):
> Cannot schedule a task before the start time of the simulation

Reply:
> since the Simulation Engine doesn't know the current time (side note - should it?)

Mmh. I remember thinking about this now. I don't know; it probably should, but I think this gets implicated alongside all the other problems with the current simulation API.